common/lib/xmodule/xmodule/tests/test_sequence.py

```diff
@@ -149,6 +157,53 @@ def test_render_student_view(self, view):
         self.assertIn("'prev_url': 'PrevSequential'", html)
         self.assertNotIn("fa fa-check-circle check-circle is-hidden", html)
 
+    @patch('xmodule.seq_module.User.objects.get', return_value=UserFactory.build())
+    def test_render_student_view_with_gating(self, mocked_user):
+        """
+        Make sure that the student view of time limited content is:
+        - checked for access if the waffle flag is on
+        - not checked for access if the waffle flag is off
+        """
+        # the order of the overrides is important since the `assert_not_called` does
+        # not appear to be limited to just the override_waffle_flag = False scope
+        with override_waffle_flag(TIMED_EXAM_GATING_WAFFLE_FLAG, active=False):
+            self._get_rendered_view(
+                self.sequence_5_1,
+                extra_context=dict(next_url='NextSequential', prev_url='PrevSequential'),
+                view=STUDENT_VIEW
+            )
+            mocked_user.assert_not_called()
+
+        with override_waffle_flag(TIMED_EXAM_GATING_WAFFLE_FLAG, active=True):
+            self._get_rendered_view(
+                self.sequence_5_1,
+                extra_context=dict(next_url='NextSequential', prev_url='PrevSequential'),
+                view=STUDENT_VIEW
+            )
+            mocked_user.assert_called_once()
+
+    @patch('xmodule.seq_module.User.objects.get', return_value=UserFactory.build())
```

Comment:
Happy to delete as I think technically the updated code doesn't affect the PUBLIC_VIEW, but it seemed to be one of the other cases tested in other unit tests here so added it in.

Reply:
If it doesn't affect the updated code, I would say we shouldn't be testing for it? But I don't really know what is this PUBLIC_VIEW exactly...